### PR TITLE
Update opentelemetry to 102.1 for security reasons.

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

This updates Opentelemetry to fix a security issue with a denial of service attack. As part of that vcenter had some breaking changes.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: docs/sources/reference/components/otelcol.receiver.zipkin.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Name | Type | Description | Default | Required
 ---- | ---- | ----------- | ------- | --------
 `parse_string_tags` | `bool` | Parse string tags and binary annotations into non-string types. | `false` | no
 `endpoint` | `string` | `host:port` to listen for traffic on. | `"0.0.0.0:9411"` | no
-`max_request_body_size` | `string` | Maximum request body size the HTTP server will allow. No limit when unset. | | no
+`max_request_body_size` | `string` | Maximum request body size the server will allow. No limit when unset. | `20MiB`         | no
```
Comment (mattdurham) [inline docs/sources/reference/components/otelcol.receiver.zipkin.md:37]: I changed the wording here to match all the others.

**Review by clayton-cornell — APPROVED** [review]
DOc chnages look OK. Over to @grafana/grafana-alloy-maintainers for code review

**Review by ptodev — COMMENTED** [review]
Thank you! Looks good, but we also need this one:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32953

**Comment by mattdurham** [comment]
@ptodev not sure what you mean by 32953.

**Comment by ptodev** [comment]
@mattdurham We need to incorporate that change into [otelcol.processor.resourcedetection](https://grafana.com/docs/alloy/latest/reference/components/otelcol.processor.resourcedetection/).

**Comment by mattdurham** [comment]
Thank you! Missed that one, will give it another look over to make sure I didnt miss anything else.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,11 +10,23 @@ internal API changes are not present.
 Main (unreleased)
 -----------------
 
+### Breaking changes
+
+- Updated OpenTelemetry to v0.102.1. (@mattdurham)
```
Comment (ptodev) [inline CHANGELOG.md:15]:
```suggestion
- Updated OpenTelemetry to v0.102.0. (@mattdurham)
```
I'd even omit the whole "Updated OpenTelemetry" line, for two reasons:
* The upgrade as a whole is not the breaking change.
* The upgrade is already mentioned below.

File: internal/component/otelcol/processor/resourcedetection/internal/azure/config.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ const Name = "azure"
 
 type Config struct {
 	ResourceAttributes ResourceAttributesConfig `alloy:"resource_attributes,block,optional"`
+	Tags               []string                 `alloy:"tags,attr,optional"`
```
Comment (ptodev) [inline internal/component/otelcol/processor/resourcedetection/internal/azure/config.go:12]: This attribute needs to be documented in the component docs.

File: CHANGELOG.md [hunk 4]
```diff
@@ -65,6 +71,9 @@ Main (unreleased)
 
 - Improved clustering use cases for tracking GCP delta metrics in the `prometheus.exporter.gcp` (@kgeckhart) 
 
+- Updated OpenTelemetry Contrib to [v0.102.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.102.0). (@mattdurham)
+  - Added `tags` to `otelcol.processor.resourcedetection`.
```
Comment (ptodev) [inline CHANGELOG.md:85]:
```suggestion
  - `otelcol.processor.resourcedetection`: Added a `tags` config argument to the `azure` detection mechanism.
  It exposes regex-matched Azure resource tags as OpenTelemetry resource attributes.
```

File: CHANGELOG.md (hunk 2 above)
Comment (mattdurham) [reply CHANGELOG.md:15]: It is OpenTelemetry v0.102.1, Contrib did not need a change so its v0.102.0.

**Review by ptodev — APPROVED** [review]

**Pull request merged**